x-pack/solutions/security/plugins/security_solution/server/lists_integration/endpoint/validators/trusted_device_validator.test.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { httpServerMock } from '@kbn/core-http-server-mocks';
+import type {
+  CreateExceptionListItemOptions,
+  UpdateExceptionListItemOptions,
+} from '@kbn/lists-plugin/server';
+import { createMockEndpointAppContextService } from '../../../endpoint/mocks';
+import { TrustedDeviceValidator } from './trusted_device_validator';
+import { ExceptionsListItemGenerator } from '../../../../common/endpoint/data_generators/exceptions_list_item_generator';
+import { EndpointArtifactExceptionValidationError } from './errors';
+
+describe('Endpoint Trusted Device API validations', () => {
+  let mockEndpointAppContextService: ReturnType<typeof createMockEndpointAppContextService>;
+  let trustedDeviceValidator: TrustedDeviceValidator;
+  let exceptionsGenerator: ExceptionsListItemGenerator;
+
+  beforeEach(() => {
+    mockEndpointAppContextService = createMockEndpointAppContextService();
+    trustedDeviceValidator = new TrustedDeviceValidator(
+      mockEndpointAppContextService,
+      httpServerMock.createKibanaRequest()
+    );
+    exceptionsGenerator = new ExceptionsListItemGenerator();
+  });
+
+  describe('Feature flag validation', () => {
+    describe('when trusted devices feature is disabled', () => {
+      beforeEach(() => {
+        (
+          mockEndpointAppContextService.experimentalFeatures as unknown as {
+            trustedDevices: boolean;
+          }
+        ).trustedDevices = false;
+      });
+
+      it('should throw error on validatePreCreateItem', async () => {
+        const item = exceptionsGenerator.generateTrustedDeviceForCreate();
+
+        await expect(
+          trustedDeviceValidator.validatePreCreateItem(
+            item as unknown as CreateExceptionListItemOptions
+          )
+        ).rejects.toThrow(EndpointArtifactExceptionValidationError);
+        await expect(
+          trustedDeviceValidator.validatePreCreateItem(
+            item as unknown as CreateExceptionListItemOptions
+          )
+        ).rejects.toThrow('Trusted devices feature is not enabled');
```

Comment:
the two calls can be merged into one:
```ts
        await expect(
          trustedDeviceValidator.validatePreCreateItem(
            item as unknown as CreateExceptionListItemOptions
          )
        ).rejects.toThrow(
          new EndpointArtifactExceptionValidationError('Trusted devices feature is not enabled')
        );
```

Reply:
but... i guess this is not a test with a long life expectancy... so this is a nit :)